Broken or Flaky test: ManagerRepairsDualAssignmentIT

**ManagerRepairsDualAssignmentIT**

**Describe the failure observed**

The ManagerRepairsDualAssignmentIT seems to be failing with some regularity.  The issue was first seen on a Jenkins run and has failed about 50% of the time running locally.

```
[ERROR] org.apache.accumulo.test.ManagerRepairsDualAssignmentIT.test -- Time elapsed: 15.79 s <<< ERROR!
java.lang.RuntimeException: java.io.EOFException: Socket is closed by peer.
        at org.apache.accumulo.core.clientImpl.TabletServerBatchReaderIterator.hasNext(TabletServerBatchReaderIterator.java:198)
        at org.apache.accumulo.server.manager.state.MetaDataTableScanner.hasNext(MetaDataTableScanner.java:125)
        at org.apache.accumulo.test.ManagerRepairsDualAssignmentIT.test(ManagerRepairsDualAssignmentIT.java:114)

```

<details>
  <summary>Full trace</summary>

```[ERROR] org.apache.accumulo.test.ManagerRepairsDualAssignmentIT.test -- Time elapsed: 15.79 s <<< ERROR!
java.lang.RuntimeException: java.io.EOFException: Socket is closed by peer.
        at org.apache.accumulo.core.clientImpl.TabletServerBatchReaderIterator.hasNext(TabletServerBatchReaderIterator.java:198)
        at org.apache.accumulo.server.manager.state.MetaDataTableScanner.hasNext(MetaDataTableScanner.java:125)
        at org.apache.accumulo.test.ManagerRepairsDualAssignmentIT.test(ManagerRepairsDualAssignmentIT.java:114)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
        at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
        at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.EOFException: Socket is closed by peer.
        at org.apache.accumulo.core.clientImpl.TabletServerBatchReaderIterator.doLookup(TabletServerBatchReaderIterator.java:918)
        at org.apache.accumulo.core.clientImpl.TabletServerBatchReaderIterator$QueryTask.run(TabletServerBatchReaderIterator.java:402)
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)

```

</details>

**Testing Environment:**
 - 2.1.3-SNAPSHOT (cac52251fd) - but started a few commits earlier.

--- a/core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchReaderIterator.java
+++ b/core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchReaderIterator.java
@@ -22,7 +22,6 @@ import static java.util.concurrent.TimeUnit.MILLISECONDS;
 import static java.util.concurrent.TimeUnit.MINUTES;
 import static java.util.concurrent.TimeUnit.SECONDS;
 
-import java.io.EOFException;
 import java.io.IOException;
 import java.time.Duration;
 import java.util.AbstractMap.SimpleImmutableEntry;
@@ -408,8 +407,7 @@ public class TabletServerBatchReaderIterator implements Iterator<Entry<Key,Value
             failures.putAll(tsFailures);
           }
         }
-      } catch (EOFException e) {
-        fatalException = e;
+
       } catch (IOException e) {
         if (!TabletServerBatchReaderIterator.this.queryThreadPool.isShutdown()) {
           synchronized (failures) {
@@ -912,13 +910,6 @@ public class TabletServerBatchReaderIterator implements Iterator<Entry<Key,Value
     } catch (TTransportException e) {
       log.debug("Server : {} msg : {}", server, e.getMessage());
       timeoutTracker.errorOccured();
-      if (e.getType() == TTransportException.END_OF_FILE) {
-        // END_OF_FILE is used in TEndpointTransport when the
-        // maxMessageSize has been reached.
-        EOFException eof = new EOFException(e.getMessage());
-        eof.addSuppressed(e);
-        throw eof;
-      }
       throw new IOException(e);
     } catch (ThriftSecurityException e) {
       log.debug("Server : {} msg : {}", server, e.getMessage(), e);

